api/proto/attest/attest.proto

```diff
@@ -42,18 +42,44 @@ message ZAttestNonceResp {
   bytes nonce = 1;      //nonce to use in quote generation
 }
 
+//Taken from
+//5.2 Crypto Agile Log Entry Format, from https://trustedcomputinggroup.org/wp-content/uploads/EFI-Protocol-Specification-rev13-160330final.pdf
+message TpmEventLogEntry {
+   uint32 index = 1;           // Sequential index of event in the log
+   uint32 pcr_index = 2;        // PCR index to which this event was measured
+   uint32 event_type = 3;       // The type of this event
+   bytes  digest256 = 4;       // The sha256 digests corresponding to this event
+   bytes  event_data_binary = 5; // The raw event data bytes
+   string event_data_string = 6; // Textual representation of the event data
+}
+
+message GPSCoordinates {
+   double latitude = 1;  // Latitude of the device geo-location
+   double longitude = 2; // Longitude of the device geo-location
+}
+
 message ZAttestQuote {
   bytes attestData = 1;  //TPMS_ATTEST (Table 2:123) in https://trustedcomputinggroup.org/wp-content/uploads/TPM-Rev-2.0-Part-2-Structures-01.38.pdf
-                          //nonce is included in attestData
+                         //nonce is included in attestData
   bytes signature = 2;   //signature to verify attestData
+  repeated bytes pcr_sha256_values = 3;     // Current values of TPM PCRs 0-15, for sha256 bank
+  repeated TpmEventLogEntry event_log = 4;  // TPM Event log
+  string eve_version = 5;                   // Current EVE version
+  string fw_version = 6;                    // Current Firmware version
+  GPSCoordinates gps_info = 7;              // Geo-Coordinates of the device
 }
 
 enum ZAttestResponseCode {
-  ATTEST_RESPONSE_NONE = 0;
-  ATTEST_RESPONSE_SUCCESS = 1;  //Attestation successful
-  ATTEST_RESPONSE_FAILURE = 2;  //Attestation failed
+  Z_ATTEST_RESPONSE_CODE_INVALID = 0;
+  Z_ATTEST_RESPONSE_CODE_SUCCESS = 1;         //Attestation successful
+  Z_ATTEST_RESPONSE_CODE_NONCE_MISMATCH = 2;  //Attestation failed due to nonce mismatch
+  Z_ATTEST_RESPONSE_CODE_QUOTE_FAILED = 3;    //Attestation failed due to quote mismatch
 }
 
 message ZAttestQuoteResp {
-  ZAttestResponseCode response = 1; //result of quote validation
+  ZAttestResponseCode response = 1;  //result of quote validation
+  bytes integrity_token = 2;         //valid if response is ATTEST_RESPONSE_SUCCESS, contains integrity token for device to
+                                     //cache and present in the next config request
+  bytes volume_storage_key = 3;      //valid if response is ATTEST_RESPONSE_SUCCESS, contains encrypted Volume Storage Key to
```

Comment:
Do you think we might end up with multiple keys over time? If so it might make sense to define this as a repeated message with a type+value.

Reply:
Updated it to have a repeated type-value pairs.